File: frontend/src/api/reservations.js
import axios from 'axios'

const API_URL = 'http://localhost:5000/reservations'

export async function reserverPanier(userId) {
  try {
    const response = await axios.post(`${API_URL}/creer`, {
      user_id: userId
    })
    return response.data
  } catch (error) {
    throw new Error(error.response?.data?.message || 'Erreur lors de la réservation.')
  }
}

export async function obtenirHistoriqueReservations(userId){
  try {
    const response = await axios.post(`${API_URL}/utilisateur//${userId}`, {
      user_id: userId
    })
    return response.data
  }catch(error){
    throw new Error(error.response?.data?.message || 'Erreur lors de la réservation.')
  }
}

File: backend/routes/reservations.py
from flask import Blueprint, request, jsonify
from database.queries.reservations_queries import creer_reservation
from database.queries.reservations_queries import obtenir_historique_reservations

reservations_bp = Blueprint("reservations", __name__)

@reservations_bp.route("/creer", methods=["POST"])
def creer_reservation_route():
    data = request.get_json()
    user_id = data.get("user_id")

    if not user_id:
        return jsonify({"message": "Champ user_id requis"}), 400

    result = creer_reservation(user_id)

    if result["success"]:
        return jsonify({"message": result["message"]}), 200
    else:
        return jsonify({"message": result["message"]}), 400

@reservations_bp.route("/utilisateur/<user_id>", methods=["GET"])
def historique_reservations(user_id):
    result = obtenir_historique_reservations(user_id)
    if result["success"]:
        return jsonify(result["reservations"]), 200
    return jsonify({"message": result["message"]}), 400
